packages/react-tinacms-github/src/github-editing-context/TinacmsGithubProvider.tsx

```diff
@@ -16,64 +16,58 @@ limitations under the License.
 
 */
 
-import React, { useState, useEffect, useCallback } from 'react'
+import React, { useState, useEffect } from 'react'
 import { useCMS } from 'tinacms'
 import GithubErrorModal from '../github-error/GithubErrorModal'
 import GithubAuthModal from './GithubAuthModal'
 import { GithubEditingContext } from './GithubEditingContext'
 import { useGithubEditing } from './useGithubEditing'
-import { authenticate as githubAuthenticate } from '../github-auth'
 import { GithubClient } from '../github-client'
 
 interface ProviderProps {
   children: any
-  clientId: string
-  authCallbackRoute: string
   editMode: boolean
   enterEditMode: () => void
   exitEditMode: () => void
   error?: any
 }
 
-interface AuthState {
-  authenticated: true
-  forkValid: true
-}
+type AuthStep = null | 'completeSetup' | 'authenticate' | 'createFork'
 
 export const TinacmsGithubProvider = ({
   children,
   editMode,
   enterEditMode,
   exitEditMode,
-  authCallbackRoute,
-  clientId,
   error: previewError,
 }: ProviderProps) => {
   const [error, setError] = useState<any>(null)
   const cms = useCMS()
   const github: GithubClient = cms.api.github
-  const [authorizingStatus, setAuthorizingStatus] = useState<AuthState | null>(
-    null
-  )
+  const [authStep, setAuthStep] = useState<AuthStep>(null)
 
   const tryEnterEditMode = async () => {
+    const validSetup =
```

Comment:
This line confuses me. Earlier we show the error modal if 'completeSetup' is true, but we set validSetup if 'completeSetup' is true

Reply:
I believe that's a mistake

Reply:
ah ok. Also I think this might be easier to read if that step is something like "invalidConfig" instead of "completeSetup"?

Reply:
This steps totally gone now